node/src/components/consensus/era_supervisor/era.rs

```diff
@@ -53,13 +54,17 @@ impl EraId {
     }
 
     /// Returns an iterator over all eras that are still bonded in this one, including this one.
-    pub(crate) fn iter_bonded(&self, bonded_eras: u64) -> impl Iterator<Item = EraId> {
-        (self.0.saturating_sub(bonded_eras)..=self.0).map(EraId)
+    pub(crate) fn iter_bonded(
+        &self,
+        first_era: EraId,
+        bonded_eras: u64,
+    ) -> impl Iterator<Item = EraId> {
+        (max(first_era.0, self.0.saturating_sub(bonded_eras))..=self.0).map(EraId)
     }
 
     /// Returns an iterator over all eras that are still bonded in this one, excluding this one.
-    pub(crate) fn iter_other(&self, count: u64) -> impl Iterator<Item = EraId> {
-        (self.0.saturating_sub(count)..self.0).map(EraId)
+    pub(crate) fn iter_other(&self, first_era: EraId, count: u64) -> impl Iterator<Item = EraId> {
+        (max(first_era.0, self.0.saturating_sub(count))..self.0).map(EraId)
```

Comment:
There's also a `max` method (but maybe you deliberately didn't use it?):
```suggestion
        (first_era.0.max(self.0.saturating_sub(count))..self.0).map(EraId)
```